Common/src/Storage/TaskLifeCycleHelper.cs

```diff
@@ -499,13 +521,51 @@ public static async Task<SessionData> ResumeAsync(ITaskTable        taskTable,
                                             .WithCancellation(cancellationToken)
                                             .ConfigureAwait(false))
     {
-      await pushQueueStorage.PushMessagesAsync(await grouping.ToListAsync(cancellationToken)
-                                                             .ConfigureAwait(false),
+      var tasks = await grouping.ToListAsync(cancellationToken)
+                                .ConfigureAwait(false);
+
+      var taskIds = tasks.Select(task => task.TaskId)
+                         .AsICollection();
+      await taskTable.UpdateManyTasks(data => data.SessionId == sessionId && data.Status == TaskStatus.Paused && taskIds.Contains(data.TaskId),
+                                      new UpdateDefinition<TaskData>().Set(data => data.Status,
+                                                                           TaskStatus.Submitted),
+                                      cancellationToken)
+                     .ConfigureAwait(false);
+      await pushQueueStorage.PushMessagesAsync(tasks,
```

Comment:
If the control plane hard crashes between these, then it can be fixed by pausing then unpausing the session